File: java/VendingMachine/src/com/ravendarque/vendingMachine/rails/RailsConfiguration.java
package com.ravendarque.vendingMachine.rails;

import java.util.*;

public class RailsConfiguration {

    private final List<RailConfigurationSettings> railConfigurationSettingsList = new ArrayList<>();

    public void add(RailConfigurationSettings settings) {

        Objects.requireNonNull(settings, "Cannot add null RailConfigurationSettings");
        railConfigurationSettingsList.add(settings);
    }

    public List<RailConfigurationSettings> getConfigurations() {

        return Collections.unmodifiableList(railConfigurationSettingsList);
    }
}

File: C++/VendingMachine/VendingMachine/RailsConfiguration.cpp
#include "pch.h"
#include "RailsConfiguration.h"
#include <list>

void RailsConfiguration::Add(const RailConfigurationSettings& railConfigurationSettings)
{
	Settings.push_back(railConfigurationSettings);
}

std::list<RailConfigurationSettings> RailsConfiguration::GetSettings() const
{
	return Settings;
}
